plugins/main/bridge/bridge.go

```diff
@@ -386,6 +454,14 @@ func setupVeth(netns ns.NetNS, br *netlink.Bridge, ifName string, mtu int, hairp
 		}
 	}
 
+	// S1031: unnecessary nil check around range (gosimple)
+	for _, v := range vlans {
```

Comment:
the default vlan will be present if the regular vlan is not set and preserveDefaultVlan is true. I would add this inside the if above (adding || len(vlans) != 0 and a if for the BridgeVlanAdd) to avoid this.

Reply:
I will suggest to move the preserveDefaultVlan condition to upper layer (not in if vlanID != 0 condition).


FROM: 
```
    if vlanID != 0 {
        if !preserveDefaultVlan {
            err = removeDefaultVlan(hostVeth)
            if err != nil {
                return nil, nil, fmt.Errorf("failed to remove default vlan on interface %q: %v", hostIface.Name, err)
            }
        }

        err = netlink.BridgeVlanAdd(hostVeth, uint16(vlanID), true, true, false, true)
        if err != nil {
            return nil, nil, fmt.Errorf("failed to setup vlan tag on interface %q: %v", hostIface.Name, err)
        }
    }
```

TO:
```
    if !preserveDefaultVlan {
        err = removeDefaultVlan(hostVeth)
        if err != nil {
            return nil, nil, fmt.Errorf("failed to remove default vlan on interface %q: %v", hostIface.Name, err)
        }
    }

    if vlanID != 0 {
        err = netlink.BridgeVlanAdd(hostVeth, uint16(vlanID), true, true, false, true)
        if err != nil {
            return nil, nil, fmt.Errorf("failed to setup vlan tag on interface %q: %v", hostIface.Name, err)
        }
    }
```

Reply:
the reasoning is that we only modify the default vlan when the vlan filtering is enabled (line 494).